File: android/app/src/main/java/com/desmoines/aipulse/ui/components/OnboardingPage.kt
package com.desmoines.aipulse.ui.components

import androidx.compose.animation.core.Spring
import androidx.compose.animation.core.animateDpAsState
import androidx.compose.animation.core.spring
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.desmoines.aipulse.ui.theme.PremiumTokens

/**
 * Single onboarding page template. Compose multiple via HorizontalPager and
 * pair with [OnboardingPageIndicator] and [OnboardingNavRow].
 * Mirrors iOS `OnboardingPage.swift`.
 */
@Composable
fun OnboardingPage(
    icon: ImageVector,
    headline: String,
    subcopy: String,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(horizontal = 32.dp, vertical = 48.dp)
            .semantics { contentDescription = "$headline. $subcopy" },
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Box(
            modifier = Modifier
                .size(140.dp)
                .clip(CircleShape)
                .background(PremiumTokens.BrandGradient),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                modifier = Modifier.size(72.dp),
                tint = androidx.compose.ui.graphics.Color.White,
            )
        }
        Spacer(Modifier.height(32.dp))
        Text(
            text = headline,
            style = MaterialTheme.typography.headlineMedium,
            fontWeight = FontWeight.ExtraBold,
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(12.dp))
        Text(
            text = subcopy,
            style = MaterialTheme.typography.bodyLarge,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
        )
    }
}

/**
 * Animated page indicator — active dot expands into a pill.
 */
@Composable
fun OnboardingPageIndicator(
    pageCount: Int,
    currentPage: Int,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier.semantics {
            contentDescription = "Page ${currentPage + 1} of $pageCount"
        },
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        repeat(pageCount) { index ->
            val isActive = index == currentPage
            val width: Dp by animateDpAsState(
                targetValue = if (isActive) 24.dp else 8.dp,
                animationSpec = spring(dampingRatio = Spring.DampingRatioMediumBouncy),
                label = "dot-width",
            )
            Box(
                modifier = Modifier
                    .height(8.dp)
                    .width(width)
                    .clip(RoundedCornerShape(4.dp))
                    .background(
                        if (isActive) MaterialTheme.colorScheme.primary
                        else MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.3f)
                    ),
            )
        }
    }
}

/**
 * Bottom nav row for onboarding with Skip + Next/Get started buttons.
 */
@Composable
fun OnboardingNavRow(
    isLastPage: Boolean,
    onSkip: () -> Unit,
    onNext: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 24.dp, vertical = 16.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        TextButton(onClick = onSkip) {
            Text("Skip")
        }
        Button(
            onClick = onNext,
            shape = RoundedCornerShape(PremiumTokens.CornerLg),
            colors = ButtonDefaults.buttonColors(
                containerColor = MaterialTheme.colorScheme.primary,
            ),
        ) {
            Text(
                text = if (isLastPage) "Get started" else "Next",
                style = MaterialTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(horizontal = 12.dp, vertical = 4.dp),
            )
        }
    }
}

/**
 * Permission priming card. Use BEFORE showing the system permission dialog
 * to explain the value of granting location or notifications.
 */
@Composable
fun PermissionPrimingCard(
    icon: ImageVector,
    title: String,
    body: String,
    primaryLabel: String,
    onPrimary: () -> Unit,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(PremiumTokens.CornerXl))
            .background(MaterialTheme.colorScheme.surface)
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Box(
            modifier = Modifier
                .size(72.dp)
                .clip(CircleShape)
                .background(MaterialTheme.colorScheme.primaryContainer),
            contentAlignment = Alignment.Center,
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                modifier = Modifier.size(36.dp),
                tint = MaterialTheme.colorScheme.onPrimaryContainer,
            )
        }
        Spacer(Modifier.height(18.dp))
        Text(
            text = title,
            style = MaterialTheme.typography.titleLarge,
            fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(8.dp))
        Text(
            text = body,
            style = MaterialTheme.typography.bodyMedium,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            textAlign = TextAlign.Center,
        )
        Spacer(Modifier.height(20.dp))
        Button(
            onClick = onPrimary,
            modifier = Modifier
                .fillMaxWidth()
                .height(48.dp),
            shape = RoundedCornerShape(PremiumTokens.CornerLg),
        ) {
            Text(
                text = primaryLabel,
                fontWeight = FontWeight.Bold,
            )
        }
        Spacer(Modifier.height(4.dp))
        TextButton(onClick = onDismiss) {
            Text("Not now")
        }
    }
}

File: ios/DesMoinesInsider/Views/Components/OnboardingPage.swift
import SwiftUI

/// Single onboarding page template. Compose multiple inside a TabView with
/// `.tabViewStyle(.page)` and pair with `OnboardingPageIndicator` +
/// `OnboardingNavRow`. Mirrors Android `OnboardingPage.kt`.
struct OnboardingPage: View {
    let systemImage: String
    let headline: String
    let subcopy: String

    var body: some View {
        VStack(spacing: 0) {
            ZStack {
                Circle()
                    .fill(PremiumTokens.brandGradient)
                    .frame(width: 140, height: 140)
                Image(systemName: systemImage)
                    .font(.system(size: 64, weight: .bold))
                    .foregroundStyle(.white)
            }

            Text(headline)
                .font(.system(size: 30, weight: .heavy))
                .multilineTextAlignment(.center)
                .padding(.top, 32)

            Text(subcopy)
                .font(.body)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
                .padding(.top, 12)
                .padding(.horizontal, 8)
        }
        .padding(.horizontal, 32)
        .padding(.vertical, 48)
        .accessibilityElement(children: .combine)
        .accessibilityLabel("\(headline). \(subcopy)")
    }
}

/// Animated page indicator — active dot expands into a pill.
struct OnboardingPageIndicator: View {
    let pageCount: Int
    let currentPage: Int

    var body: some View {
        HStack(spacing: 8) {
            ForEach(0..<pageCount, id: \.self) { index in
                let isActive = index == currentPage
                Capsule()
                    .fill(isActive ? Color.accentColor : Color.secondary.opacity(0.3))
                    .frame(width: isActive ? 24 : 8, height: 8)
                    .animation(
                        .spring(response: 0.4, dampingFraction: 0.65),
                        value: currentPage
                    )
            }
        }
        .accessibilityLabel("Page \(currentPage + 1) of \(pageCount)")
    }
}

/// Bottom nav row for onboarding with Skip + Next / Get started buttons.
struct OnboardingNavRow: View {
    let isLastPage: Bool
    let onSkip: () -> Void
    let onNext: () -> Void

    var body: some View {
        HStack {
            Button("Skip", action: onSkip)
                .foregroundStyle(.secondary)
            Spacer()
            Button {
                HapticFeedback.shared.light()
                onNext()
            } label: {
                Text(isLastPage ? "Get started" : "Next")
                    .font(.headline)
                    .padding(.horizontal, 20)
                    .padding(.vertical, 12)
                    .background(Color.accentColor)
                    .foregroundStyle(.white)
                    .clipShape(RoundedRectangle(cornerRadius: PremiumTokens.cornerLg))
            }
            .buttonStyle(.plain)
        }
        .padding(.horizontal, 24)
        .padding(.vertical, 16)
    }
}

/// Permission priming card shown BEFORE the system permission dialog so users
/// understand the value before being prompted.
struct PermissionPrimingCard: View {
    let systemImage: String
    let title: String
    let body: String
    let primaryLabel: String
    let onPrimary: () -> Void
    let onDismiss: () -> Void

    var body: some View {
        VStack(spacing: 0) {
            ZStack {
                Circle()
                    .fill(Color.accentColor.opacity(0.15))
                    .frame(width: 72, height: 72)
                Image(systemName: systemImage)
                    .font(.system(size: 30, weight: .semibold))
                    .foregroundStyle(Color.accentColor)
            }

            Text(title)
                .font(.title3.bold())
                .multilineTextAlignment(.center)
                .padding(.top, 18)

            Text(body)
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
                .padding(.top, 8)

            Button {
                HapticFeedback.shared.medium()
                onPrimary()
            } label: {
                Text(primaryLabel)
                    .font(.headline)
                    .frame(maxWidth: .infinity)
                    .frame(height: 48)
                    .background(Color.accentColor)
                    .foregroundStyle(.white)
                    .clipShape(RoundedRectangle(cornerRadius: PremiumTokens.cornerLg))
            }
            .buttonStyle(.plain)
            .padding(.top, 20)

            Button("Not now", action: onDismiss)
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .padding(.top, 4)
        }
        .padding(24)
        .background(Color(.systemBackground))
        .clipShape(RoundedRectangle(cornerRadius: PremiumTokens.cornerXl))
        .premiumShadow(elevation: 8)
    }
}

#Preview {
    VStack {
        OnboardingPage(
            systemImage: "sparkles",
            headline: "Discover Des Moines",
            subcopy: "Find events, restaurants, and hidden gems curated just for you."
        )
        OnboardingPageIndicator(pageCount: 4, currentPage: 1)
        OnboardingNavRow(isLastPage: false, onSkip: {}, onNext: {})
    }
}
